### PR TITLE
MARP-59 migrate the marketplace from ivyteam jenkins to GitHub

### PR DESCRIPTION
**Pull request by tutn-axonivy**

### CODE REVIEW

File: src/app/pages/_error/404.twig [hunk 1]
```diff
@@ -7,13 +7,16 @@
 {% block content %}
   <section id="one">
     <div class="inner">
-      <h1 class="text-center"><i class="si si-404"></i> Process start not found...</h1>
+      <h1 class="text-center">
+        <i class="si si-404"></i>
+        Process start not found...</h1>
```
Comment (ntqdinh-axonivy) [inline src/app/pages/_error/404.twig:12]: h1 closing tag should in the next line after format code, right?
(and the P tag of Reason below also)

File: src/web/_market/a-trust/meta.json [hunk 2]
```diff
@@ -17,5 +17,6 @@
             "artifactId": "a-trust-connector-product",
             "type": "zip"
         }
-    ]    
+    ]
 }
+
```
Comment (ntqdinh-axonivy) [inline src/web/_market/a-trust/meta.json]: remove this redundant line

File: src/web/_market/docusign-connector/meta.json [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
+    {
+      "repoUrl": "https://maven.axonivy.com",
+      "name": "DocuSign Connector Product",
+      "groupId": "com.axonivy.connector.docusign",
+      "artifactId": "docusign-connector-product",
+      "type": "zip"
+    }
+  ]
+}
+
```
Comment (ntqdinh-axonivy) [inline src/web/_market/docusign-connector/meta.json:22]: remove this line

File: src/web/_market/quick-start-tutorial/meta.json [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
+  "mavenArtifacts": [
+    {
+      "name": "Quick Start Tutorial Product",
+      "groupId": "com.axonivy.demo",
+      "artifactId": "quick-start-tutorial-product",
+      "type": "zip"
+    }
+  ]	
 }
+
```
Comment (ntqdinh-axonivy) [inline src/web/_market/quick-start-tutorial/meta.json:23]: remove this line

File: src/web/_market/sftp-connector/meta.json [hunk 5]
```diff
@@ -19,3 +19,4 @@
         }
     ]    
 }
+
```
Comment (ntqdinh-axonivy) [inline src/web/_market/sftp-connector/meta.json:22]: remove this line

File: src/web/_market/tel-search-ch-connector/meta.json [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
+    {
+      "repoUrl": "https://maven.axonivy.com",
+      "name": "tel.search.ch Connector Product",
+      "groupId": "com.axonivy.connector.telsearch",
+      "artifactId": "tel-search-ch-connector-product",
+      "type": "zip"
+    }
+  ]
 }
+
```
Comment (ntqdinh-axonivy) [inline src/web/_market/tel-search-ch-connector/meta.json:22]: remove this line

File: src/web/_market/twitter/meta.json [hunk 7]
```diff
@@ -1,23 +1,24 @@
...
+    {
+      "repoUrl": "https://maven.axonivy.com",
+      "name": "Twitter Connector Product",
+      "groupId": "com.axonivy.connector.twitter",
+      "artifactId": "twitter-connector-product",
+      "type": "zip"
+    }
+  ]
 }
+
```
Comment (ntqdinh-axonivy) [inline src/web/_market/twitter/meta.json:24]: remove this line

File: src/web/_market/ui-path-connector/meta.json [hunk 8]
```diff
@@ -1,21 +1,22 @@
...
+    {
+      "repoUrl": "https://maven.axonivy.com",
+      "name": "UiPath Connector Product",
+      "groupId": "com.axonivy.connector.uipath",
+      "artifactId": "ui-path-connector-product",
+      "type": "zip"
+    }
+  ]	
+}
+
```
Comment (ntqdinh-axonivy) [inline src/web/_market/ui-path-connector/meta.json:22]: remove this line

File: src/app/domain/market/Product.php [hunk 9]
```diff
@@ -339,6 +339,7 @@ private function assetBaseUrl_versionized(string $version): string
   {
     $key = $this->product->getKey();
     $artifactId = $this->product->getProductArtifactId();
+    echo "/market-cache/$key/$artifactId/$version";
```
Comment (ntqdinh-axonivy) [inline src/app/domain/market/Product.php]: remove echo

File: src/web/assets/js/util.js [hunk 10]
```diff
@@ -1,37 +1,39 @@
...
-      b = [];
+    ($a = $this.find("a")), (b = []);
 
     $a.each(function () {
-
       var $this = $(this),
-        indent = Math.max(0, $this.parents('li').length - 1),
-        href = $this.attr('href'),
-        target = $this.attr('target');
+        indent = Math.max(0, $this.parents("li").length - 1),
```
Comment (ntqdinh-axonivy) [inline src/web/assets/js/util.js]: js can understand singlequote as string

**Review by ntqdinh-axonivy — CHANGES_REQUESTED** [review]

File: src/web/_market/a-trust/meta.json (hunk 2 above)
Comment (tutn-axonivy) [reply src/web/_market/a-trust/meta.json]: all the new lines was added to fix editor checker

**Review by ntqdinh-axonivy — APPROVED** [review]
LGTM

**Comment by ivy-rew** [comment]
Hi @tutn-axonivy 
Cool that you start to build pipelines on github. Note that the jenkins-pipeline is still in charge though ... builds are started on PRs automatically ... so if there is an error and you merge anyway, we have red builds to analyze.
https://github.com/axonivy-market/market.axonivy.com/pull/31

**Comment by ivy-rew** [comment]
As you are doing first steps in this repo I think you should let the work review by the initial contributors. Especially @alexsuter in this repo. 
This also applies to other market-repos too ... just involve the main contributor on PRs.

**Comment by alexsuter** [comment]
I like that you have formatted the code with .editorconfig. But please if you make such reformattings then do this in an own PR. So that we see that this PR is only a formatting PR and should not change the behavior.
You made 108 commits and merged it to master. This makes the history unreadable and unusable. Such PRs should be squashed before your merge.


Do we have any behavior changes in this PR?

**Comment by azjustin1** [comment]
It doesn't contain any change in behavior but adds a GitHub actions script. But because the editor config scan makes the runner fail every time. So I decided to update the format of the code to make it pass. Sorry for not notifying that.

**Comment by alexsuter** [comment]
You changed the Dockerfile which is currently in charge.

**Comment by tutn-axonivy** [comment]
I did some tricks to make it run in my local. I'm so sorry for not discarding it when committing the code.

**Comment by ivy-rew** [comment]
> I did some tricks to make it run in my local. I'm so sorry for not discarding it when committing the code.

No problem; we're glad you're contributing after all. :handshake: 
Let's learn to get better on the way: maybe you could make a retro topic out this story, to establish contribution guidelines within team Octopus?

**Pull request merged**